docs/GOVERNANCE.md

```diff
@@ -0,0 +1,207 @@
+- [Background](#background)
+
+## Two-Tier System
+
+*Classification system distinguishing officially supported components from community contributions.*
+
+### Core Tier
+
+**Officially supported components** maintained by the Kubeflow community.
+
+**Requirements:**
+- 90% test coverage (unit, integration, e2e)
+- Security review passed
+- Complete documentation
+- Active maintenance commitment
+- Backward compatibility guarantees
+
+**Benefits:**
+- Official support and maintenance
+- Included in releases
+- Priority for bug fixes
+- Long-term stability guarantees
+
+### Third-Party Tier
+
+**Community-contributed components** with lighter requirements.
+
+**Requirements:**
+- 80% test coverage
+- Basic documentation (README, examples)
+- At least 2 maintainers
+- Working implementation
+
+**Benefits:**
+- Community visibility
+- Shared maintenance burden
+- Faster contribution process
+- Innovation sandbox
+
+## Ownership Models
+
+*How ownership, maintenance, and decision-making responsibilities are distributed across tiers.*
+
+### Core Tier
+- **Owned by**: Kubeflow community
+- **Maintained by**: Designated maintainer teams
+- **Decisions by**: Steering committee consensus
+- **Support**: Official community support
+
+### Third-Party Tier
+- **Owned by**: Original contributors
+- **Maintained by**: Component owners
+- **Decisions by**: Component owners
+- **Support**: Best-effort community support
+
+## Tier Transitions
+
+*Process for moving components between Core and Third-Party tiers.*
+
+### Promotion to Core
+1. **Nomination**: Any maintainer can nominate
+2. **Review**: Technical and governance review
+3. **Requirements**: Must meet all core tier requirements
+4. **Decision**: Steering committee approval
+5. **Timeline**: 4-6 weeks review process
+
+### Demotion from Core
+Triggers:
+- Maintenance neglect (>6 months)
+- Security issues unaddressed
+- Breaking changes without migration
+- Community consensus
+
+## Removal Policies
+
+*Timeline and criteria for removing inactive or problematic components from the repository.*
+
+### Verification Process (9 months)
+Components are marked for verification if:
+- No updates in 9 months
+- Maintainers unresponsive
+- Compatibility issues
+
+### Removal Process (12 months)
+After 12 months of inactivity:
+1. **Notice**: 30-day removal notice
+2. **Community input**: 2-week feedback period
+3. **Final decision**: Steering committee
+4. **Removal**: Delete component code from repository
+
+### Emergency Removal
+Immediate removal for:
+- Critical security vulnerabilities
+- Legal issues
+- Malicious code
+
+## Deprecation Policy
+
+*Structured approach to deprecating components with adequate notice and migration support.*
+
+### Two-Release Policy
+Components deprecated for 2 Kubeflow releases before removal.
+
+**Process:**
+1. **Deprecation notice**: Mark as deprecated
+2. **Migration guide**: Provide alternatives
+3. **Community notice**: Announce in releases
```

Comment:
hmm sounds like we could miss this type of announcement during releases, are we planning to automate this?